### PR TITLE
NMS-15634: Docs section about startup configuration and opennms.conf

### PR DESCRIPTION
**Pull request by Bonrob2**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15634

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+= Startup Configuration
+
+There are a number of startup settings that you can configure, including `JAVA_HEAP_SIZE`.
+{page-component-title} startup scripts source `\$OPENNMS_HOME/etc/opennms.conf` early in their startup, if that file exists.
+
+No `opennms.conf` exists in the default RPM/DEB installation, but `\$OPENNMS_HOME/etc/examples` includes a sample file that you can use as a starting point.
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:7]:
```suggestion
There are a number of startup settings that you can configure, including `JAVA_HEAP_SIZE`.
{page-component-title} startup scripts load `\${OPENNMS_HOME}/etc/opennms.conf` early in their startup, if that file exists.

No `opennms.conf` exists in the default RPM/DEB installation, but `\${OPENNMS_HOME}/etc/examples` includes a sample file that you can use as a starting point.
```

**Comment by Bonrob2** [comment]
> We probably should have a little more description on each, specifically with heap size, to give some guidance for why these should be tweaked.

Sounds good. I'll see what I can find.

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 2]
```diff
@@ -6,14 +6,46 @@ There are a number of startup settings that you can configure, including `JAVA_H
...
+| Variable
+| Description
+| Default
+| Behavior
+
+| `JAVA_HEAP_SIZE`
+| Specifies the maximum amount of memory the JVM should use.
+You may want to increase this value if you experience long response times, frequent garbage collection that takes a lot of time, or see alarms that should have been cleared automatically still listed as alarms.
+See https://opennms.discourse.group/t/performance-tuning/1438[performance tuning] on Discourse for more information.
+| 512 MB
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:24]: Current default:
```suggestion
| 2048
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 3]
```diff
@@ -6,14 +6,46 @@ There are a number of startup settings that you can configure, including `JAVA_H
...
+
+| `ADDITIONAL_MANAGER_OPTIONS`
+| Additional options to pass to Java when starting {page-component-title}.
+|
+| Appended to other JVM startup flags.
+
+| `MAXIMUM_FILE_DESCRIPTORS`
+| You may want to change these values if you end up with too many "files open exceptions" in {page-component-title}.
+See https://opennms.discourse.group/t/how-to-investigate-file-descriptor-issues/953[How to investigate file descriptor issues] on Discourse for more information.
+| 204800
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:44]: default per https://github.com/OpenNMS/opennms/blob/foundation-2023/opennms-base-assembly/src/main/filtered/bin/opennms
```suggestion
| 20480
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 4]
```diff
@@ -6,14 +6,46 @@ There are a number of startup settings that you can configure, including `JAVA_H
...
+| Units are in MB.
+Suffix `m` is added automatically with the result passed to JVM as an `-Xmx`.
+
+|`JAVA_INITIAL_HEAP_SIZE`
+| Specifies the initial amount of memory the JVM should use.
+Effectively sets a minimum memory value when used in conjunction with `JAVA_HEAP_SIZE`.
+You may want to increase this value if you experience long response times, frequent garbage collection that takes a lot of time, or see alarms that should have been cleared automatically still listed as alarms.
+It is good practice to set the `JAVA_INITIAL_HEAP_SIZE` and `JAVA_HEAP_SIZE` to the same value.
+| Units are in MB.
+| Suffix `m` is added automatically with the result passed to JVM as an `-Xms` flag.
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:34]:
```suggestion
| null
| Units are in megabytes.
Suffix `m` is added automatically with the result passed to JVM as an `-Xms` flag.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 5]
```diff
@@ -6,14 +6,46 @@ There are a number of startup settings that you can configure, including `JAVA_H
...
+| `MAXIMUM_FILE_DESCRIPTORS`
+| You may want to change these values if you end up with too many "files open exceptions" in {page-component-title}.
+See https://opennms.discourse.group/t/how-to-investigate-file-descriptor-issues/953[How to investigate file descriptor issues] on Discourse for more information.
+| 204800
+| Passed to `ulimit -n` before starting JVM.
+
+| `MAXIMUM_SIZE_STACK_SEGMENT`
+| Defines the size limit for each Java thread in the JVM.
+| 8192 kbytes
+| Passed to `ulimit -s` before starting JVM.
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:50]:
```suggestion
| 8192
| Units are in kilobytes.
Passed to `ulimit -s` before starting JVM.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc [hunk 6]
```diff
@@ -6,14 +6,46 @@ There are a number of startup settings that you can configure, including `JAVA_H
...
+| Description
+| Default
+| Behavior
+
+| `JAVA_HEAP_SIZE`
+| Specifies the maximum amount of memory the JVM should use.
+You may want to increase this value if you experience long response times, frequent garbage collection that takes a lot of time, or see alarms that should have been cleared automatically still listed as alarms.
+See https://opennms.discourse.group/t/performance-tuning/1438[performance tuning] on Discourse for more information.
+| 512 MB
+| Units are in MB.
```
Comment (mmahacek) [inline docs/modules/operation/pages/deep-dive/admin/configuration/startup.adoc:25]:
```suggestion
| Units are in megabytes.
```

**Review by mmahacek — APPROVED** [review]

**Pull request merged**